src/hotspot/share/opto/escape.cpp

```diff
@@ -1033,10 +1045,12 @@ void ConnectionGraph::updates_after_load_split(Node* data_phi, Node* previous_lo
       // The base might not be something that we can create an unique
       // type for. If that's the case we are done with that input.
       PointsToNode* jobj_ptn = unique_java_object(base);
-      if (jobj_ptn == nullptr || !jobj_ptn->scalar_replaceable()) {
+      if (base->is_Phi() && !reducible_merges.member(base)) {
+        continue;
+      }
+      if (!base->is_Phi() && (jobj_ptn == nullptr || !jobj_ptn->scalar_replaceable())) {
```

Comment:
is_Phi() can be called only once by putting this case in an else statement. The tradeoff is more lines of code vs 2 seprate calls to is_Phi(). I don't how how significant this is, just an observation.

Reply:
Thanks, Modified to

if (base->is_Phi() && !reducible_merges.member(base)) {
        continue;
      } else if (jobj_ptn == nullptr || !jobj_ptn->scalar_replaceable()) {
        continue;
  }

Reply:
Thanks, Modified to
      if (base->is_Phi()) {
        if (!reducible_merges.member(base)) {
          continue;
        }
      } else {
        if (jobj_ptn == nullptr || !jobj_ptn->scalar_replaceable()) {
          continue;
        }
      }